style_editor.html

```diff
@@ -9391,87 +9403,99 @@
-      if (name.startsWith('EFFECT_GAME')) {
-        gameEffectsMenu.appendChild(option);
-      } else if (name.startsWith('EFFECT_USER')) {
-        userEffectsMenu.appendChild(option);
-      } else if ([
-          EFFECT_STUN, 
-          EFFECT_FIRE, 
-          EFFECT_CLIP_IN, 
-          EFFECT_CLIP_OUT, 
-          EFFECT_RELOAD, 
-          EFFECT_MODE, 
-          EFFECT_RANGE, 
-          EFFECT_EMPTY, 
-          EFFECT_FULL, 
-          EFFECT_JAM, 
-          EFFECT_UNJAM, 
-          EFFECT_PLI_ON, 
-          EFFECT_PLI_OFF
-        ].includes(Number(value))) {
-        blasterEffectsMenu.appendChild(option);
-      } else {
-        generalEffectsMenu.appendChild(option);
-      }
-      actions[value] = function() { blade.addEffect(value, 0.0) };
-    }
-  }
+  <script>
+         // Get the menu and button elements
+        const menu = FIND('more_effects_menu');
+        const button = FIND('do_selected');
+
+        // Create an object to store actions for each item
+        const actions = {};
+
+        /* Add values from the enum builder to an array and sort alphabetically,
+        excluding effects with dedicated buttons.*/
+        const values = Object.entries(EFFECT_ENUM_BUILDER.value_to_name)
+          .sort((a, b) => a[1].localeCompare(b[1]))
+          .filter(([value]) => ![
+            EFFECT_CLASH,
+            EFFECT_STAB,
+            EFFECT_BLAST,
+            EFFECT_FORCE,
+            EFFECT_BOOT,
+            EFFECT_NEWFONT,
+            EFFECT_PREON,
+          ].includes(Number(value)));
+
+        // Add sub-groups for the different categories of effects
+        const generalEffectsMenu = document.createElement('optgroup');
+        generalEffectsMenu.label = 'General Effects';
+        const userEffectsMenu = document.createElement('optgroup');
+        userEffectsMenu.label = 'User Effects';
+        const blasterEffectsMenu = document.createElement('optgroup');
+        blasterEffectsMenu.label = 'Blaster Effects';
+        const gameEffectsMenu = document.createElement('optgroup');
+        gameEffectsMenu.label = 'Game Effects';
+        const errorMessagesMenu = document.createElement('optgroup');
+        errorMessagesMenu.label = 'Error Messages';
+
+        // Add sorted values to the menu and actions dictionary,
+        for (const [value, name] of values) {
+          const nameWithoutEffect = name.replace(/^EFFECT_/, '');
+          const option = document.createElement('option');
+          option.value = value;
+          option.text = nameWithoutEffect;
+
+          // Check if the effect belongs to a certain category and add it to the corresponding sub-group
+          if (name.startsWith('EFFECT_GAME')) {
+            gameEffectsMenu.appendChild(option);
+          } else if (name.startsWith('EFFECT_USER')) {
+            userEffectsMenu.appendChild(option);
+          } else if ([
+              EFFECT_STUN, 
```

Comment:
For the rest of these I think we should just use a switch statement.

Reply:
ok. how about now